use_cases/eluc/app/app.py

```diff
@@ -1,118 +1,49 @@
 """
-Main app file for ELUC demo.
-Uses many 'components' to separate divs and their related callbacks.
-They aren't necessarily truly reusable components, but they help to organize the code.
+Main entrypoint to run the app. Contains the layout of the app and registers all the callbacks of each component.
 """
-import pandas as pd
-from dash import Dash
-from dash import dcc
-from dash import html
+from dash import Dash, html
 import dash_bootstrap_components as dbc
+import pandas as pd
 
 import app.constants as app_constants
-from app.components.chart import ChartComponent
-from app.components.legend import LegendComponent
-from app.components.lock import LockComponent
-from app.components.map import MapComponent
-from app.components.prediction import PredictionComponent
-from app.components.prescription import PrescriptionComponent
+from app.components.intro import IntroComponent
+from app.components.context.context import ContextComponent
+from app.components.filter import FilterComponent
+from app.components.dms.dms import DMSComponent
 from app.components.references import ReferencesComponent
-from app.components.sliders import SlidersComponent
-from app.components.trivia import TriviaComponent
+from app.utils import EvolutionHandler
+
 
 app = Dash(__name__,
            external_stylesheets=[dbc.themes.BOOTSTRAP, dbc.icons.BOOTSTRAP],
            prevent_initial_callbacks="initial_duplicate")
 server = app.server
 
-df = pd.read_csv(app_constants.DATA_FILE_PATH, index_col=app_constants.INDEX_COLS)
-
-legend_component = LegendComponent()
-
-map_component = MapComponent(df)
-map_component.register_update_map_callback(app)
-map_component.register_click_map_callback(app)
-map_component.register_select_country_callback(app)
-
-prescription_component = PrescriptionComponent(df)
-prescription_component.register_select_prescriptor_callback(app)
-prescription_component.register_toggle_modal_callback(app)
-
-sliders_component = SlidersComponent(df)
-sliders_component.register_set_frozen_reset_sliders_callback(app)
-sliders_component.register_show_slider_value_callback(app)
-sliders_component.register_sum_to_one_callback(app)
-
-lock_component = LockComponent()
-
-chart_component = ChartComponent(df)
-chart_component.register_update_context_chart_callback(app)
-chart_component.register_update_presc_chart_callback(app)
+app.title = 'Land Use Optimization'
 
-prediction_component = PredictionComponent(df)
-prediction_component.register_predictor_callback(app)
-prediction_component.register_land_use_callback(app)
+app_df = pd.read_csv("app/data/app_data.csv", index_col=app_constants.INDEX_COLS)
 
-trivia_component = TriviaComponent(df)
-trivia_component.register_update_trivia_callback(app)
+evolution_handler = EvolutionHandler()
 
+intro_component = IntroComponent()
+context_component = ContextComponent(app_df, evolution_handler)
+filter_component = FilterComponent(evolution_handler)
+dms_component = DMSComponent(app_df, evolution_handler)
```

Comment:
We pass our app df and evolution handler to the components so that they have the ability to prescribe, look at context, etc.